pkg/registry/core/pod/storage/eviction.go

```diff
@@ -291,14 +292,19 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 	}
 
 	// Try the delete
-	err = addConditionAndDeletePod(r, ctx, eviction.Name, rest.ValidateAllObjectFunc, deleteOptions)
+	err = addConditionAndDeletePod(r, ctx, pod, deleteOptions)
 	if err != nil {
-		if errors.IsConflict(err) && updateDeletionOptions &&
-			(originalDeleteOptions.Preconditions == nil || originalDeleteOptions.Preconditions.ResourceVersion == nil) {
-			// If we encounter a resource conflict error, we updated the deletion options to include them,
-			// and the original deletion options did not specify ResourceVersion, we send back
-			// TooManyRequests so clients will retry.
-			return nil, createTooManyRequestsError(pdbName)
+		// retry is useful if we made a decision on a stale information
+		if errors.IsConflict(err) && updateDeletionOptions {
+			// check that the original preconditions do not conflict with a pod resourceVersion or UID
+			hasConflictingOriginalPreconditions := originalDeleteOptions.Preconditions != nil && (originalDeleteOptions.Preconditions.ResourceVersion != nil ||
+				(originalDeleteOptions.Preconditions.UID != nil && *originalDeleteOptions.Preconditions.UID != pod.UID))
+			if !hasConflictingOriginalPreconditions {
```

Comment:
I think we should revert this change, for a couple reasons:
1. we only want clients to retry on a resourceVersion conflict, a uid conflict means this is a completely different pod
2. this is too difficult to reason about if we want to backport this